### ISSUE TITLE
Update subjective featureset for Memories

### ISSUE THREAD

**Issue by pulsejet** [meichthys/foss_photo_libraries#30]
Wonder if [memories](https://github.com/pulsejet/polaroid) belongs here; could use some better visibility. Basically we've have feature parity with Nextcloud Photos now, except a couple of minor things.

**Comment by meichthys** [comment]
I agree! I'm just not sure how to present it. Does it get its own column? If so do the Nextcloud mobile apps count for the mobile features. I'm trying to figure out the best way to accurately include it in the list.

**Comment by pulsejet** [comment]
Not sure! It does leverage the most important part of the Nextcloud mobile apps, i.e. auto upload. But the UI is still unfortunately separate (albeit very responsive).

**Comment by meichthys** [comment]
@pulsejet I've added Memories. If you have any suggested changes to my suggested scores, please let me know.

**Comment by pulsejet** [comment]
Thanks!

**Comment by pulsejet** [comment]
@meichthys I just happened to look at this list again and do have some questions. I mainly compared Memories to Immich since they're closest in terms of feature set.

Please note that I don't intend to complain; just trying to understand the reasoning behind the scoring.

1. Video support - How does Immich get 8/10 without transcoding support? AFAIK you can't even play e.g. HEVC on Immich in the browser yet (https://github.com/immich-app/immich/issues/203), neither is streaming adaptive. Although support for Memories is new, Photoprism still gets 7/10 despite supporting transcoding since a while?
2. Object/face recognition: Immich has the same score as Memories/Photoview/Nextcloud Photos despite not supporting face recognition at all? (https://github.com/immich-app/immich/issues/294)
3. Timeline: Immich 9/10, Memories 8/10. Why? IMHO (at least in the webapp) the memories timeline is much better. E.g. try skipping to a past position in the timeline in both apps and scroll up; immich (at least in the demo) is quite janky for me while memories is very smooth. Also unless I missed something Immich shows square thumbs only.
4. Multiple user support: anything based on Nextcloud has way better support for this compared to anything else. Nextcloud integrates with many auth frameworks for "real" multi-user support.
5. EXIF Data: I'm a bit confused why Memories has 3/10. What important feature is it missing that other apps support?
6. Automatic upload: Trying to figure out what the Nextcloud app is missing in uploads. I've been using it for a few months now and haven't run into any issues whatsoever.

In general, I feel the list is a little biased in favor of Immich. But that's just my opinion.

**Comment by meichthys** [comment]
@pulsejet Thanks for your input. My scores are primarily based off the Demos that each project provides. I understand that the demos may not be fully representative of all features of the project, resource usage, etc, but that's really all i have to go on since i don't have the hardware or time to run an self-hosted instance of each. I'm glad to change the subjective measure based off of input like yours. 
(I try not to be as fair as possible even though I currently use Memories as my primary photo library 😉. Perhaps that makes me be a bit more aware or critical of the Memories deficiencies)

1. Video support was primarily based on my experience in the demos, not necessarily on transcoding. The Immich scoring was primarily based off the mobile app (not the browser). The Immich mobile app didn't seem to have any troubles playing any of the media files - I'm not sure how it does it without transcoding. 🤷‍♂️ I'm on memories v4.7.2 and still can't seem to play most videos. It may be a configuration issue on my end, but if it does require involved configuration, then I think that should result in a lower subjective score. ([I haven't been able to use the Memories demo to test this](https://github.com/pulsejet/memories/issues/142), so i'm reliant on my own instance)
2. You are definitely correct here. I was looking at this yesterday and realized that the Immich rating is probably too high. I think it does support object detection, but not face detection yet. (perhaps object/face detection should be broken out into separate features). The memories 8/10 rating was probably prior to some of your recent changes that improved the functionality. I've made an [issue here for the immich adjustment.](https://github.com/meichthys/foss_photo_libraries/issues/32)
3. I think the immich demo is currently down but i will test this. Again, I was using the mobile app for rating this feature since that's where the majority of Immich useres interact with the photos (IMO, the mobile app is significantly better than the web app - although it has been getting close to feature parity in recent releases).
4. Again, you are correct on this for sure. Not sure why it was a 7 on Nextcloud/Memories to start. [Will fix shortly.](https://github.com/meichthys/foss_photo_libraries/issues/33)
5. The Memories rating is old. This should be increased (pull request would be appreciated!). Other apps do allow modification to different exif fields (more than just the date), but like you indicated, 7 for immich is probably a stretch.
6. This can also probably be increased for Nextclou/Memories now. Before this week, the ios nextcloud mobile app required the app to be open for the photos to upload (i think the android app may work better). Either way, there are alot of peopleon [android](https://github.com/nextcloud/android/issues?q=is%3Aissue+is%3Aopen+auto+upload) and [ios](https://github.com/nextcloud/ios/issues?q=is%3Aissue+is%3Aopen+auto+upload) that complain about the nextcloud auto-uploads). For me the uploads now work automatically in the background on iOS since the latest v4.5.5.2 nextcloud ios update.

As for being biased toward Immich, it may be due to the fact that Immich has submitted their own pull request to this repo (Unless I have a real good reason to bump the subjective score down, i generally leave it as is to encourage more pull requests.) Also, the mobile app experience which is quite polished may help the user gloss over some deficiencies at first glance.

In summary, I'm the one who ends up having to give a subjective rating and i realize that means I'm probably the bad guy to one project or the other. I'm just a single tester trying to give an accurate as possible view of the options that are out there, so input like yours helps a lot.

**Comment by pulsejet** [comment]
> @pulsejet Thanks for your input. My scores are primarily based off the "out of the box" experience that the Demos for each project provides. I understand that the demos may not be fully representative of all features of the project, resource usage, etc, but that's really all i have to go on since i don't have the hardware or time to run an updated, properly-configured, self-hosted instance of each.

Yup, that's my source of knowledge as well. Btw, I realised I didn't mention thanks for maintaining this in the first place 😄 

> 1. Video support was primarily based on my experience in the demos, not necessarily on transcoding. The Immich scoring was primarily based off the mobile app (not the browser). The Immich mobile app didn't seem to have any troubles playing any of the media files - I'm not sure how it does it without transcoding. 🤷‍♂️ I'm on memories v4.7.2 and still can't seem to play most videos. It may be a configuration issue on my end, but if it does require involved configuration, then I think that should result in a lower subjective score. ([I haven't been able to use the Memories demo to test this](https://github.com/pulsejet/memories/issues/142), so i'm reliant on my own instance)

Native apps don't have many limitations unless you're running older hardware or your bandwidth bottlenecks (then you need adaptive streaming like memories does). I agree transcoding is a lot of extra config (suggest filing an issue if it isn't working), but that's unfortunately thanks to Intel and NVIDIA not making life any easier 😞. Apps like Jellyfin get it easier when deployed with Docker since they're geared towards media in the first place (Nextcloud isn't, yet) so they can include some non-free components in the image.
As for the demo, that instance isn't going to do any transcoding anyway 😄

> 3. I think the immich demo is currently down but i will test this. Again, I was using the mobile app for rating this feature since that's where the majority of Immich useres interact with the photos (IMO, the mobile app is significantly better than the web app - although it has been getting close to feature parity in recent releases).

I didn't mean to say the Immich timeline has problems. All I'm trying to understand is why Memories has a lower score. (IMHO, Memories timeline is as good as Google Photos if not a bit faster).

To be fair, some of my evaluation points are from the POV of a developer rather than a user. For example, a justified layout in a timeline view is _the_ most complicated part of the Memories codebase (but nowhere close to the most useful). Having square thumbnails is much easier since it lets you predict where each photo will be simply by knowing the total number of photos in each section. As soon as the size of thumbnails becomes variable, this turns into a very complex problem. Which is exactly why no other FOSS photo library (that I know) has a timeline view supporting both scrubbing _and_ a justified layout simultaneously.

> 6. This can also probably be increased for Nextclou/Memories now. Before this week, the ios nextcloud mobile app required the app to be open for the photos to upload (i think the android app may work better).

I don't know why stuff keeps breaking in the Nextcloud apps honestly ... It's not like the WebDAV API has changed (at all) 😶. Just that 4 seemed a bit low. I do use Android btw.

> As for being biased toward Immich, it may be due to the fact that Immich has submitted their own pull request to this repo (Unless I have a real good reason to bump the subjective score down, i generally leave it as is to encourage more pull requests.) Also, the mobile app experience which is quite polished may help the user gloss over some deficiencies at first glance.
> 
> In summary, I'm the one who ends up having to give a subjective rating and i realize that means I'm probably the bad guy to one project or the other. I'm just a single tester trying to give an accurate as possible view of the options that are out there, so input like yours helps a lot.

I don't blame anyone for the bias I felt; as I said it's just my opinion anyway. Just wanted to point out some of the things (e.g. face recognition and exif) where it seemed not obvious to me why Immich had a high score.

**Comment by meichthys** [comment]
@pulsejet keep up the great work! I've updated the subjective measures to bring things up to date for Nextcloud/Memories. Let me know if you have any other suggestions, and please feel free to submit pull requests when/if changes are made.
